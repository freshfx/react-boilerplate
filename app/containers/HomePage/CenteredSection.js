import styled from '@emotion/styled'

import Section from './Section'

const CenteredSection = styled(Section)`
  text-align: center;
`

export default CenteredSection
