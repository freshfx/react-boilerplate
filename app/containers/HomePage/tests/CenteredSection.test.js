import React from 'react'
import {shallow} from 'enzyme'

import CenteredSection from '../CenteredSection'

describe('<CenteredSection />', () => {
  it('should have a className attribute', () => {
    const renderedComponent = shallow(<CenteredSection />)
    expect(renderedComponent.prop('className')).toBeDefined()
  })

  it('should adopt a valid attribute', () => {
    const id = 'test'
    const renderedComponent = shallow(<CenteredSection id={id} />)
    expect(renderedComponent.prop('id')).toEqual(id)
  })
})
