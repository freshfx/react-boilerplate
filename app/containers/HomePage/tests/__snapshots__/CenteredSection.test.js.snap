// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CenteredSection /> should match snapshot 1`] = `
.c0 {
  margin: 3em auto;
  text-align: center;
}

.c0:first-child {
  margin-top: 0;
}

<CenteredSection>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "CenteredSection-sc-6zy0wx-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "margin:3em auto;&:first-child{margin-top:0;}",
            "text-align:center;",
          ],
        },
        "displayName": "CenteredSection",
        "foldedComponentIds": Array [
          "Section-sc-1eulrsm-0",
        ],
        "render": [Function],
        "styledComponentId": "CenteredSection-sc-6zy0wx-0",
        "target": "section",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <section
      className="c0"
    />
  </StyledComponent>
</CenteredSection>
`;
